### PR TITLE
Add "signup" / create user in authentication-jwt module

### PR DESCRIPTION
**Pull request by HrithikMittal**
This pull request is to provide the signup method in the authentication-jwt So it would be easier for developers.
https://github.com/strongloop/loopback-next/blob/master/extensions/authentication-jwt/src/__tests__/fixtures/controllers/user.controller.ts

Fixes https://github.com/strongloop/loopback-next/issues/5312

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [ ] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: extensions/authentication-jwt/src/__tests__/fixtures/controllers/user.controller.ts [hunk 1]
```diff
@@ -95,4 +95,39 @@ export class UserController {
...
+          },
+        },
+      },
+    },
+  })
+  async signUp(
+    @requestBody({
+      content: {
+        'application/json': {
+          schema: getModelSchemaRef(NewUserRequest, {
```
Comment (madaky) [inline extensions/authentication-jwt/src/__tests__/fixtures/controllers/user.controller.ts:116]: NO import

Comment (madaky) [inline extensions/authentication-jwt/src/__tests__/fixtures/controllers/user.controller.ts:116]: No definition for NewUserRequest

File: extensions/authentication-jwt/src/__tests__/fixtures/controllers/user.controller.ts [hunk 2]
```diff
@@ -95,4 +95,39 @@ export class UserController {
...
+            title: 'NewUser',
+          }),
+        },
+      },
+    })
+    newUserRequest: NewUserRequest,
+  ): Promise<User> {
+    const password = await hash(newUserRequest.password, await genSalt());
+    const savedUser = await this.userRepository.create(
```
Comment (madaky) [inline extensions/authentication-jwt/src/__tests__/fixtures/controllers/user.controller.ts:125]: Will throw a property does not exist. as no dependency injection.

File: extensions/authentication-jwt/src/__tests__/fixtures/controllers/user.controller.ts [hunk 3]
```diff
@@ -95,4 +95,39 @@ export class UserController {
...
+        'application/json': {
+          schema: getModelSchemaRef(NewUserRequest, {
+            title: 'NewUser',
+          }),
+        },
+      },
+    })
+    newUserRequest: NewUserRequest,
+  ): Promise<User> {
+    const password = await hash(newUserRequest.password, await genSalt());
```
Comment (madaky) [inline extensions/authentication-jwt/src/__tests__/fixtures/controllers/user.controller.ts:124]: No definition for hash and gen salt

File: extensions/authentication-jwt/src/__tests__/fixtures/controllers/user.controller.ts [hunk 4]
```diff
@@ -95,4 +95,39 @@ export class UserController {
...
+    const savedUser = await this.userRepository.create(
+      _.omit(newUserRequest, 'password'),
```
Comment (madaky) [inline extensions/authentication-jwt/src/__tests__/fixtures/controllers/user.controller.ts:126]: failed to import lodash

Comment (dougal83) [reply extensions/authentication-jwt/src/__tests__/fixtures/controllers/user.controller.ts:126]: While we are here, try to remove the dependency on lodash.  `delete newUserRequest.password` might be suitable here or possible use destructing?

Comment (dougal83) [reply extensions/authentication-jwt/src/__tests__/fixtures/controllers/user.controller.ts:126]: > I think we have lodash in package.json so might we can use this. But @dougal83 your suggestion helped me some way out there. 

@madaky I believe that the general intent is to remove dependencies such as lodash over time where possible as they are not entirely necessary.

File: extensions/authentication-jwt/src/__tests__/fixtures/controllers/user.controller.ts (hunk 1 above)
Comment (jannyHou) [reply extensions/authentication-jwt/src/__tests__/fixtures/controllers/user.controller.ts:116]: @HrithikMittal you can find `NewUserRequest` here
https://github.com/strongloop/loopback-next/blob/master/examples/todo-jwt/src/controllers/user.controller.ts#L22-L29

File: extensions/authentication-jwt/src/__tests__/fixtures/controllers/user.controller.ts (hunk 3 above)
Comment (jannyHou) [reply extensions/authentication-jwt/src/__tests__/fixtures/controllers/user.controller.ts:124]: They are imported from `bcryptjs`:
https://github.com/strongloop/loopback-next/blob/master/examples/todo-jwt/src/controllers/user.controller.ts#L19

**Comment by HrithikMittal** [comment]
Hi, Everyone Sorry to create an another pull request for this but I do the changes there and also test it in my local laptop. I think it would work Please look at this https://github.com/strongloop/loopback-next/pull/5763
and Now I am closing this PR.
Thank you everyone.

**Pull request closed without merging**